packages/completer-extension/schema/tracker.json

```diff
@@ -0,0 +1,24 @@
+{
+  "title": "Code Completion",
+  "description": "Code Completion settings.",
+  "jupyter.lab.setting-icon-label": "Code Completion settings",
+  "properties": {
+    "availableProviders": {
```

Comment:
As this is purely informative, a better way would be to not add this properties and to dynamically update the description of `selectedProviders` to include a list of available values. You can look at how this is done for listing the command in shortcuts: https://github.com/jupyterlab/jupyterlab/blob/master/packages/shortcuts-extension/src/index.ts#L51

Let me know if you need help to understand that piece of code. This is a tricky part; but the idea is to define a transformation of the settings (using `registry.transform(<id>, callbacks)`) between the value obtained from the server and what the user will see.

Reply:
Thanks @fcollonval, indeed this `transform` method is what I'm looking for!